dbos/_event_loop.py

```diff
@@ -58,10 +63,23 @@ async def _shutdown(self) -> None:
         await asyncio.gather(*tasks, return_exceptions=True)
         self._loop.stop()
 
+    def set_main_loop(self):
```

Comment:
Should we call `set_main_loop` upon `enqueue_async` as well? I look at some of our test code, we call `DBOS.launch()` first and then `uvicorn.run()` -- the event loop is created upon `uvicorn.run`.

Reply:
It's also called in the start method here, for cases where you `DBOS.launch()` from the main thread

Reply:
I deliberately don't want to handle that case--can get very confusing very fast, especially if you have multiple threads for whatever reason. The documented behavior will be "use the event loop in which DBOS was launched (with a special case for FastAPI)"